### PR TITLE
New default index page

### PR DESCRIPTION
**Pull request by LazeMSS**
Improved the default page with a full guide how to setup the slices
Changed the OctoPrint API response to the latest version of OP

I know it uses bootstrap and a lot of external includes - for me it seems the logical choice but we could also finde a css framework that is light and include this directly.

### CODE REVIEW

**Comment by spuder** [comment]
So much better, thank you for doing this. 

Screen Shot 2023-04-04 at 2 46 44 PM

### Ideas: 
1. What if there is more than 1 printer? I suspect these instructions will need to change to `127.0.0.1:4470/<printerid>` (should be addressed in a future PR)
2. What if we mirrored the green highlighting that ankerslicer uses? 
Screen Shot 2023-04-04 at 2 51 25 PM

**Comment by LazeMSS** [comment]
1. Yes when we know how to handle multiple printers it is easy because we already have that data and I will implement a dropdown for selecting the printer and the guide will show the right one - we haven't decided on how each printer should be implemented yet - but the webui is easy to fix and the template stuff is easy to do.
2. Ohh instead of the blue one - i just used the blue from the bootstrap themes so it would match later on if using the standard themes.

File: static/index.html [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		<meta charset="utf-8">
+		<meta content="width=device-width, initial-scale=1" name="viewport">
+		<title>ankerctl</title>
+		<link crossorigin="anonymous" href="//cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha3/dist/css/bootstrap.min.css" integrity="sha384-KK94CHFLLe+nY2dmCWGMq91rCGa5gtU4mk92HdvYe+M/SXH301p5ILy+dN9+nJOZ" rel="stylesheet">
+	</head>
+	<body>
+		<div class="container my-5">
+			<div class="p-2 text-center bg-body-tertiary rounded-3">
+				<h1 class="text-body-emphasis display-3">ankerctl</h1>
+				<p class="lead">Congratulations you have ankerctl running&hellip;</p>
```
Comment (spuder) [inline static/index.html:13]:
```suggestion
				<p class="lead">Congratulations on running ankerctl&hellip;</p>
```

File: static/index.html [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+			</div>
+		</div>
+		<div class="container mb-4">
+			<div class="row row-cols-1 row-cols-md-1 row-cols-lg-2 row-cols-xxl-2 g-3 p-2">
+				<div class="col-12 col-lg-12">
+					<h2 class="display-6 fs-2 mb-2">Connecting PrusaSlicer/SuperSlicer</h2>
+					<ol class="list-group list-group-numbered">
+						<li class="list-group-item">A simple default list group item</li>
+						<li class="list-group-item">Go to "Printer settings" tab</li>
+						<li class="list-group-item">Click "Gear"-icon (Add/Edit Physical Printer)</li>
```
Comment (spuder) [inline static/index.html:23]:
```suggestion
						<li class="list-group-item">Click "Gear"⚙️ (Add/Edit Physical Printer)</li>
```

File: static/index.html [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+						<li class="list-group-item">Fill in "Descriptive name for the printer" with whatever you like</li>
+						<li class="list-group-item">Select your Ankermake printer profile from the dropdown menu</li>
+						<li class="list-group-item">Under "Host type" select "OctoPrint"</li>
+						<li class="list-group-item">In "Hostname, IP or URL" fill in with: <kbd>{{ configHost }}:{{ configPort }}</kbd><a  class="text-decoration-none" href="#" id="configData">📋</a></li>
```
Comment (spuder) [inline static/index.html:27]:
```suggestion
						<li class="list-group-item">In "Hostname, IP or URL" fill in with: <kbd>{{ configHost }}:{{ configPort }}</kbd><a  class="text-decoration-none" href="#" id="configData">📋</a></li>
						<li class="list-group-item">In "Leave APIKey/ Password Blank"</a></li>
```

File: static/index.html [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+				<h1 class="text-body-emphasis display-3">ankerctl</h1>
+				<p class="lead">Congratulations you have ankerctl running&hellip;</p>
+			</div>
+		</div>
+		<div class="container">
+			<div class="row row-cols-1 row-cols-md-1 row-cols-lg-2 row-cols-xxl-2 g-3 p-2">
+				<div class="col-12 col-lg-12">
+					<h2 class="display-6 fs-2 mb-2">Connecting PrusaSlicer/SuperSlicer</h2>
+					<ol class="list-group list-group-numbered">
+						<li class="list-group-item">A simple default list group item</li>
```
Comment (thomasjpatterson) [inline static/index.html:21]: Suggest removing this line.

File: static/index.html [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+						<li class="list-group-item">Click "Gear"-icon (Add/Edit Physical Printer)</li>
+						<li class="list-group-item">Fill in "Descriptive name for the printer" with whatever you like</li>
+						<li class="list-group-item">Select your Ankermake printer profile from the dropdown menu</li>
+						<li class="list-group-item">Under "Host type" select "OctoPrint"</li>
+						<li class="list-group-item">In "Hostname, IP or URL" fill in with: <kbd>{{ configHost }}:{{ configPort }}</kbd><a  class="text-decoration-none" href="#" id="configData">📋</a></li>
+						<li class="list-group-item">Click "Test" to validate input</li>
+						<li class="list-group-item">Click "Ok"</li>
+						<li class="list-group-item">Go and slice yor files</li>
+						<li class="list-group-item">Press "G&#x3E;" button to start direct upload</li>
+						<li class="list-group-item">Press "Upload and print" - "Upload" is not supported</li>
```
[the remark this answers is not in the document]
Comment (LazeMSS) [reply static/index.html:32]: The wording on the button is "Upload" and not "Upload only" that is why i wrote it like that

File: static/prusaslicer-setup-3.png
Comment (chrivers) [inline static/prusaslicer-setup-3.png:1]: Shouldn't we have a blue highlight for the "G>" button here?

File: static/prusaslicer-setup.png
Comment (chrivers) [inline static/prusaslicer-setup.png:1]: Can we please call this `prusaslicer-setup-1.png`, for consistency? :D

File: static/index.html [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+				<div class="col-12 col-lg-12 text-center">
+					<img src="static/prusaslicer-setup.png" class="img-thumbnail" alt="step 1">
+					<img src="static/prusaslicer-setup-2.png" class="img-thumbnail mt-2" alt="step 2">
+					<img src="static/prusaslicer-setup-3.png" class="img-thumbnail mt-2" alt="step 3">
+					<img src="static/prusaslicer-setup-4.png" class="img-thumbnail mt-2" alt="step 4">
+				</div>
+			</div>
+			{{ configData }}
+		</div>
+		<script crossorigin="anonymous" integrity="sha384-ENjdO4Dr2bkBIFxQpeoTz1HIcje39Wm4jDKdf19U8gI4ddQ3GYNS7NTKfAdVQSZe" src="//cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha3/dist/js/bootstrap.bundle.min.js"></script>
```
Comment (chrivers) [inline static/index.html:45]: For most sites, I'm a big fan of CDNs, but here I'm not so sure.

It's not unreasonable for people to want to use this project where there is little or no internet connectivity.

How much space would it take to vendor a copy of these in, say, `/static/vendor/...` ?

File: static/index.html [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+		<script>
+			$(function () {
+				$('#configData').on('click',function(){
+					navigator.clipboard.writeText("{{ configHost }}:{{ configPort }}");
+					return false;
+				})
+			});
+		</script>
+	</body>
+</html>
```
Comment (chrivers) [inline static/index.html:56]: Missing newline at end of file :)

File: static/index.html [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+		<meta charset="utf-8">
+		<meta content="width=device-width, initial-scale=1" name="viewport">
+		<title>ankerctl</title>
+		<link crossorigin="anonymous" href="//cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha3/dist/css/bootstrap.min.css" integrity="sha384-KK94CHFLLe+nY2dmCWGMq91rCGa5gtU4mk92HdvYe+M/SXH301p5ILy+dN9+nJOZ" rel="stylesheet">
+	</head>
+	<body>
+		<div class="container my-5">
+			<div class="p-2 text-center bg-body-tertiary rounded-3">
+				<h1 class="text-body-emphasis display-3">ankerctl</h1>
+				<p class="lead">Congratulations on running ankerctl&hellip;</p>
```
Comment (chrivers) [inline static/index.html:13]: It feels a bit odd to have "..." at the end of a headline. But an exclamation mark would also be too much, I think. Maybe we just remove `&hellip;`?

Comment (LazeMSS) [inline static/index.html:13]:
```suggestion
				<p class="lead">Congratulations on running ankerctl</p>
```

**Comment by spuder** [comment]
I've got a broken link on the first image

Screen Shot 2023-04-04 at 3 56 32 PM

**Comment by LazeMSS** [comment]
> I've got a broken link on the first image
> 
> Screen Shot 2023-04-04 at 3 56 32 PM

fixed

File: static/index.html [hunk 9]
```diff
@@ -0,0 +1,56 @@
+<!DOCTYPE html>
+<html data-bs-theme="dark" lang="en">
+	<head>
+		<meta charset="utf-8">
+		<meta content="width=device-width, initial-scale=1" name="viewport">
+		<title>ankerctl</title>
+		<link href="static/vendor/bootstrap-5.3.0-alpha3-dist/css/bootstrap.min.css" rel="stylesheet">
```
Comment (chrivers) [inline static/index.html:7]: As discussed, let's use `/static/...` instead of `static/...`

**Review by chrivers — APPROVED** [review]
Looks good! Merging :)

Thanks for your contribution.

By the way, next time, please try to avoid merge commits in the middle of the history, it makes it more difficult than necessary to follow. Rebasing on top of master is much preferred :)

**Pull request merged**